nodestream_akamai/property/property.py

```diff
@@ -2,39 +2,36 @@
 
 from nodestream.pipeline.extractors import Extractor
 
-from ..akamai_utils.iam_client import AkamaiIamClient
 from ..akamai_utils.property_client import AkamaiPropertyClient
 
 
 class AkamaiPropertyExtractor(Extractor):
     def __init__(self, **akamai_client_kwargs) -> None:
         self.client = AkamaiPropertyClient(**akamai_client_kwargs)
-        self.iam_client = AkamaiIamClient(**akamai_client_kwargs)
         self.logger = logging.getLogger(self.__class__.__name__)
 
     async def extract_records(self):
+        self.logger.debug("extracting records")
         try:
             properties = self.client.list_all_properties()
         except Exception as err:
-            self.logger.error("Failed to list properties: %s", err)
-            return
+            self.logger.exception("Failed to list properties: %s", err)
+            raise err
```

Comment:
`logger.exception("msg")` is equivalent to `logger.error("msg", exc_info=True)` and is the preferred way to include stack trace and other info into the log.